src/App.tsx

```diff
@@ -1,27 +1,75 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
+import { Button } from './components/Button';
+import { GoodsList } from './components/GoodsList';
+import { Loader } from './components/Loader';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [currentClickedButton, setCurrentClickedButton] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleFetchData = useCallback(
+    async (callback: () => Promise<Good[]>, type: string) => {
+      if (currentClickedButton !== type) {
```

Comment:
I want to have the possibility to click on this button if it has an error to retry